apps/system/js/app_chrome.js

```diff
@@ -340,6 +372,68 @@
     window.dispatchEvent(new CustomEvent('global-search-request'));
   };
 
+  AppChrome.prototype.onClickSiteIcon = function onClickSiteIcon() {
+    if (!this.app.isBrowser() || this._loading) {
```

Comment:
Nit: In general, I'm trying to avoid duplicating state in AppChrome and AppWindow in case they get out of sync. I think it would be better to read this directly from the AppWindow object with this.app.loading. It isn't a big deal in this case because I think the only other thing that sets AppWindow.loading is AppWindow.destroyBrowser() - it probably just means that AppChrome thinks that a crashed iframe is still loading and won't let you pin the page.